### ISSUE TITLE
Need Help

### ISSUE THREAD

**Issue by Eton** [leafinity/keras_ganomaly#1]
hi, I have a question need your help.

In your demo code

why np.expand_dims(new_x, axis=-1) / 127 - 1 in function reshape_x

please help me 

thanks

**Comment by Tassfighter** [comment]
Haven't looked at the actual images but my guess:

1. Keras needs an image as an array of height / width / channels. The MNIST grey-scale image is just a 2d array with shape (height, width) and has to be expanded to shape (height, width, 1). A color image would have (height, width, 3) or (3, height, width) if it's Theano instead of TensorFlow

2. The color space is from 0 to 255 (uint). Image / 127 - 1 converts the color space from -1 to +1 (the auto-encoders output is also -1 to +1). BTW Image / 127 - 1 should be  Image / 127.5 - 1 instead (255 / 2 = 127.5 not 127)

**Comment by forechoandlook** [comment]
> Haven't looked at the actual images but my guess:
> 
> 1. Keras needs an image as an array of height / width / channels. The MNIST grey-scale image is just a 2d array with shape (height, width) and has to be expanded to shape (height, width, 1). A color image would have (height, width, 3) or (3, height, width) if it's Theano instead of TensorFlow
> 2. The color space is from 0 to 255 (uint). Image / 127 - 1 converts the color space from -1 to +1 (the auto-encoders output is also -1 to +1). BTW Image / 127 - 1 should be  Image / 127.5 - 1 instead (255 / 2 = 127.5 not 127)

Obviously, you are right! We call this operation as the preprocessing of data.
